modules/nextflow/src/main/groovy/nextflow/processor/PublishDir.groovy

```diff
@@ -321,6 +336,11 @@ class PublishDir {
         }
         catch( Throwable e ) {
             log.warn "Failed to publish file: ${source.toUriString()}; to: ${target.toUriString()} [${mode.toString().toLowerCase()}] -- See log file for details", e
+            if( NF.strictMode || failOnError){
+                if( taskProcessor && taskRun ){
+                    taskProcessor.publishFailed(taskRun, e, target)
```

Comment:
The session can be accessed directly using this `Global.session` singleton. An example here 

https://github.com/nextflow-io/nextflow/blob/master/modules/nextflow/src/main/groovy/nextflow/processor/PublishDir.groovy#L106-L106

Reply:
yes, I saw we can access to session using Global but I think is better to delegate the logic to who call us (taskProcessor)

Reply:
fair enough